e2e/testinfra/terraform/modules/service_account/service_account.tf

```diff
@@ -24,15 +24,50 @@ resource "google_service_account" "gcp_sa" {
 }
 
 resource "google_service_account_iam_member" "k8s_sa_binding" {
+  for_each = toset([
```

Comment:
Do we really want to be managing these with terraform?
It might be better to manage them with K8s RBAC in the e2e framework. That way it auto-applies and can be test-specific.

Reply:
I tried that approach earlier in a different PR, but then got pushed back because we don't want to grant `e2e-test-runner` permissions to create IAM bindings.

Reply:
Why does it need IAM tho? Can't they just be RBAC?

Reply:
These IAM bindings allow R*Sync KSAs to be able to impersonate a GSA at the project level.

We can't mange them with K8s RBAC at the GKE cluster level.

Reply:
If we really want them to be auto-applied and test-specific, we have to create the IAM bindings in the test. Then the prow job (`e2e-test-runner`) needs to have privileged permissions to create IAM bindings, which we want to avoid.

Reply:
oh, I see. These are for workloadIdentityUser, not cluster permissions. My mistake.

It seems unfortunate that we need to modify this TF and manually apply it any time we want to test in a new namespace or with a new RootSync.

Maybe we should have a pool of these so we don't need so many IAM bindings...

Reply:
This means we're giving all the RootSyncs and RepoSyncs permission to impersonate the same GSA, with whatever IAM that GSA has. That's not really how customers are going to use WI...

It might be safer to follow the principle of least privilege and only give each test the permissions it needs. Tho that would mean giving the test GSA permission to modify IAM in the project.

Reply:
Personally, I think giving the test GSA IAM to modify IAM is probably fine, as long as it's only scoped to the test project. It already has permission to do all sorts of other expensive things. Giving it project admin won't make it appreciably more risky at this point.

Reply:
resolving because discussion is in the other thread